browsers/playwright-execution.mdx

```diff
@@ -0,0 +1,247 @@
+```python Python
+from kernel import Kernel
+
+kernel = Kernel()
+
+# Create a browser
+kernel_browser = kernel.browsers.create()
+
+# Execute Playwright code
+response = kernel.browsers.playwright.execute(
+    id=kernel_browser.session_id,
+    code="""
+        await page.goto('https://example.com')
+        return await page.title()
+    """
+)
+
+print(response.result)  # "Example Domain"
+```
+
+```bash CLI
+kernel browsers playwright execute <session_id> 'await page.goto("https://www.onkernel.com"); return page.title();'
+```
+</CodeGroup>
+
+## Available variables
+
+Your code has access to these Playwright objects:
+
+- `page` - The current page instance
+- `context` - The browser context
+- `browser` - The browser instance
+
+## Returning values
+
+Use a `return` statement to send data back from your code:
+
+<CodeGroup>
+```typescript Typescript/Javascript
+const response = await kernel.browsers.playwright.execute(
+  sessionId,
+  {
+    code: `
+      await page.goto('https://example.com');
+      const title = await page.title();
+      const url = page.url();
+      return { title, url };
+    `
+  }
+);
+
+console.log(response.result); // { title: "Example Domain", url: "https://example.com" }
+```
+
+```python Python
+response = kernel.browsers.playwright.execute(
+    id=session_id,
+    code="""
+        await page.goto('https://example.com')
+        title = await page.title()
+        url = page.url()
+        return {'title': title, 'url': url}
+    """
+)
+
+print(response.result)  # {'title': 'Example Domain', 'url': 'https://example.com'}
+```
+</CodeGroup>
+
+## Timeout configuration
+
+Set a custom timeout (default is 60 seconds, max is 300 seconds):
+
+<CodeGroup>
+```typescript Typescript/Javascript
+const response = await kernel.browsers.playwright.execute(
+  sessionId,
+  {
+    code: `
+      await page.goto('https://example.com');
+      return await page.title();
+    `,
+    timeout_sec: 120
+  }
+);
+```
+
+```python Python
+response = kernel.browsers.playwright.execute(
+    id=session_id,
+    code="""
+        await page.goto('https://example.com')
+        return await page.title()
+    """,
+    timeout_sec=120
+)
+```
+</CodeGroup>
+
+## Error handling
+
+The response includes error information if execution fails:
+
+<CodeGroup>
+```typescript Typescript/Javascript
+const response = await kernel.browsers.playwright.execute(
+  sessionId,
+  {
+    code: `
+      await page.goto('https://invalid-url');
+      return await page.title();
+    `
+  }
+);
+
+if (!response.success) {
+  console.error('Error:', response.error);
+  console.error('Stderr:', response.stderr);
+}
+```
+
+```python Python
+response = kernel.browsers.playwright.execute(
+    id=session_id,
+    code="""
+        await page.goto('https://invalid-url')
+        return await page.title()
+    """
+)
+
+if not response.success:
+    print('Error:', response.error)
+    print('Stderr:', response.stderr)
+```
+</CodeGroup>
+
+## Use cases
```

Comment:
What do you think about adding an additional use case explaining how to take screenshots via this method? @masnwilliams